extensions/mssql/src/objectExplorerNodeProvider/objectExplorerNodeProvider.ts

```diff
@@ -10,83 +10,60 @@ import * as vscode from 'vscode';
 	public readonly supportedProviderId: string = constants.providerId;
-	private sessionMap: Map<string, Session>;
+	private sessionMap: Map<string, SqlClusterSession>;
 	private expandCompleteEmitter = new vscode.EventEmitter<sqlops.ObjectExplorerExpandInfo>();
 
 	constructor(private appContext: AppContext) {
 		super();
-
-		this.sessionMap = new Map();
-		this.appContext.registerService<MssqlObjectExplorerNodeProvider>(constants.ObjectExplorerService, this);
+		this.sessionMap = new Map<string, SqlClusterSession>();
+		this.appContext.registerService<SqlObjectExplorerNodeProvider>(constants.ObjectExplorerService, this);
 	}
 
-	handleSessionOpen(session: sqlops.ObjectExplorerSession): Thenable<boolean> {
+	handleSessionOpen(sqlSession: sqlops.ObjectExplorerSession): Thenable<boolean> {
 		return new Promise((resolve, reject) => {
-			if (!session) {
+			if (!sqlSession) {
 				reject('handleSessionOpen requires a session object to be passed');
 			} else {
-				resolve(this.doSessionOpen(session));
+				resolve(this.doSessionOpen(sqlSession));
 			}
 		});
 	}
 
-	private async doSessionOpen(sessionInfo: sqlops.ObjectExplorerSession): Promise<boolean> {
-		let connectionProfile = await sqlops.objectexplorer.getSessionConnectionProfile(sessionInfo.sessionId);
-		if (!connectionProfile) {
-			return false;
-		} else {
-			let credentials = await sqlops.connection.getCredentials(connectionProfile.id);
-			let serverInfo = await sqlops.connection.getServerInfo(connectionProfile.id);
-			if (!serverInfo || !credentials || !serverInfo.options) {
-				return false;
-			}
-			let endpoints: IEndpoint[] = serverInfo.options[constants.clusterEndpointsProperty];
-			if (!endpoints || endpoints.length === 0) {
-				return false;
-			}
-			let index = endpoints.findIndex(ep => ep.serviceName === constants.hadoopKnoxEndpointName);
-			if (index === -1) {
-				return false;
-			}
+	private async doSessionOpen(sqlSession: sqlops.ObjectExplorerSession): Promise<boolean> {
+		if (!sqlSession && !sqlSession.sessionId) { return false; }
 
-			let connInfo: sqlops.connection.Connection = {
-				options: {
-					'host': endpoints[index].ipAddress,
-					'groupId': connectionProfile.options.groupId,
-					'knoxport': endpoints[index].port,
-					'user': 'root', //connectionProfile.options.userName cluster setup has to have the same user for master and big data cluster
-					'password': credentials.password,
-				},
-				providerName: constants.mssqlClusterProviderName,
-				connectionId: UUID.generateUuid()
-			};
-
-			let connection = new Connection(connInfo);
-			connection.saveUriWithPrefix(constants.objectExplorerPrefix);
-			let session = new Session(connection, sessionInfo.sessionId);
-			session.root = new RootNode(session, new TreeDataContext(this.appContext.extensionContext, this), sessionInfo.rootNode.nodePath);
-			this.sessionMap.set(sessionInfo.sessionId, session);
-			return true;
-		}
+		let sqlSessionId = sqlSession.sessionId;
```

Comment:
Why create a variable you only use once? Can we just pass session.sessionId into the `getSessionConnectionProfile` method?